Tests/differ.py

```diff
@@ -196,6 +197,9 @@ def _paths_are_same_kind(path1, path2):
 
 
 def _validate_path(path_str):
+    if path_str is None:
+        raise argparse.ArgumentTypeError(
+            "{} is not a valid path.".format(path_str))
```

Comment:
I don't think this validation is needed because this method will only be called whent the command line argument is provided; if it's not provided things will fail earlier

Reply:
@miguelsousa   I put the code in because there is a path where this not caught. In runnner.run_test(), if either of the subprocess calls encounters an exception, it is caught and the function returns None as the output path. Most of the test functions receive this as 'actual_path', and then call differ().  In differ.main(), the get_options() call converts the "None" to the current directory. The result is a mis-leading error message that the path types differ, as the expected_path is a file, and the actual_path is a directory.

Reply:
OK, in that case please add a test case for this situation.